resources/shared/test-invoker.mjs

```diff
@@ -31,7 +33,58 @@ export class RAFTestInvoker extends TestInvoker {
     }
 }
 
+class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
+    static mc = new MessageChannel();
+    _scheduleCallbacks(resolve) {
+        let gotTimer = false;
+        let gotMessage = false;
+        let gotPromise = false;
+
+        const tryTriggerAsyncCallback = () => {
+            if (!gotTimer || !gotMessage || !gotPromise)
+                return;
+
+            this._asyncCallback();
+            setTimeout(async () => {
+                await this._reportCallback();
+                resolve();
+            }, 0);
+        };
+
+        requestAnimationFrame(async () => {
+            await this._syncCallback();
+            gotPromise = true;
+            tryTriggerAsyncCallback();
+        });
+
+        requestAnimationFrame(() => {
+            setTimeout(async () => {
+                await Promise.resolve();
+                gotTimer = true;
+                tryTriggerAsyncCallback();
+            });
+
+            AsyncRAFTestInvoker.mc.port1.addEventListener(
+                "message",
+                async function () {
+                    await Promise.resolve();
+                    gotMessage = true;
+                    tryTriggerAsyncCallback();
+                },
+                { once: true }
+            );
+            AsyncRAFTestInvoker.mc.port1.start();
+            AsyncRAFTestInvoker.mc.port2.postMessage("speedometer");
+        });
+    }
+}
+
 export const TEST_INVOKER_LOOKUP = {
     __proto__: null,
     raf: RAFTestInvoker,
 };
+
+export const ASYNC_TEST_INVOKER_LOOKUP = {
```

Comment:
Do we really need a separate lookup table? Now that timer-based measurement is gone,
why don't we just have a single lookup table with "raf" and "async" as measurement methods.